flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -910,14 +910,28 @@ private ApplicationReport startAppMaster(
         }
 
         // Upload and register user jars
-        final List<String> userClassPaths =
+        List<String> userClassPaths =
```

Comment:
Because we need to check if usrlib exists later. If usrlib exists, we need `userClassPaths.addAll(usrLibClassPaths);` so we can reuse current code to make other options o YarnConfigOptions.UserJarInclusion work as expected.

Reply:
Even though we could still have the `final`.

Reply:
Fix in 986fff4543fbfd5016f087c53533b395af8c166f